### ISSUE TITLE
AWS ECR auth doesn't work while proxy on cache miss

### ISSUE THREAD

**Issue by Tefnet** [enix/kube-image-keeper#141]
When image is not found in cache ("cached image is not available, proxying origin"), proxy fails to auth correctly to ECR.
E0804 13:50:46.022737       1 server.go:159] could not proxy registry: invalid character 'N' looking for beginning of value
The response from ECR is "Not Authorized" while proxy tries to parse JSON.

The problem is only during direct proxy to ECR on cache-miss, at the same time ECR auth works correctly for image caching (it is eventually cached)

**Comment by Nicolasgouze** [comment]
Hi @Tefnet ,we worked recently on the AWS ECR topic (and pushed new releases correcting a part of the problems we used to have), can you please let us know which release you used for your tests ? Do you still repro the issue with 1.3.0 Beta 1 or 1.3.0 Beta 2 ? We'll shortly generate a new (non beta) release.

**Comment by donch** [comment]
Hi @Tefnet , would you mind provide more details : 
- Your Kubernetes location (inside or outside AWS)
- Which auth method do you use to pull your images from ECR ? 
- Your image name pattern 

Thanks

**Comment by Tefnet** [comment]
@donch hi,

- EKS (k8s v1.27)
- IRSA
- [account_id].dkr.ecr.[region].amazonaws.com/project/service:[branch_name]-[short_sha]-[timestamp]

here are the logs from kube-image-keeper-proxy pod (redacted repository hostname and project/service name, but they're correct in original logs)

```
I1018 07:09:38.881563       1 server.go:138] "proxying request" repository="project/service" originRegistry="012345678.dkr.ecr.us-east-2.amazonaws.com"
I1018 07:09:38.925926       1 server.go:141] "cached image is not available, proxying origin" originRegistry="012345678.dkr.ecr.us-east-2.amazonaws.com" error="404 Not Found"
I1018 07:09:38.925947       1 server.go:230] "listing CachedImages" repositoryLabel="012345678.dkr.ecr.us-east-2.amazonaws.com-project-service"
E1018 07:09:39.422074       1 server.go:155] could not proxy registry: invalid character 'N' looking for beginning of value
[GIN] 2023/10/18 - 07:09:39 | 500 |  540.613771ms |     10.76.36.24 | HEAD     "/v2/012345678.dkr.ecr.us-east-2.amazonaws.com/project/service/manifests/branch_name-short_sha-timestamp"
Error #01: invalid character 'N' looking for beginning of value
```
